modules/axhal/src/platform/pc_x86/acpi.rs

```diff
@@ -0,0 +1,333 @@
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_ptr();
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.read_volatile() }
+    }
+
+    fn read_pci_u16(&self, segment: u16, bus: u8, device: u8, function: u8, offset: u16) -> u16 {
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_ptr() as *const u16;
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.read_volatile() }
+    }
+
+    fn read_pci_u32(&self, segment: u16, bus: u8, device: u8, function: u8, offset: u16) -> u32 {
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_ptr() as *const u32;
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.read_volatile() }
+    }
+
+    fn write_pci_u8(
+        &self,
+        segment: u16,
+        bus: u8,
+        device: u8,
+        function: u8,
+        offset: u16,
+        value: u8,
+    ) {
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_mut_ptr();
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.write_volatile(value) }
+    }
+
+    fn write_pci_u16(
+        &self,
+        segment: u16,
+        bus: u8,
+        device: u8,
+        function: u8,
+        offset: u16,
+        value: u16,
+    ) {
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_mut_ptr() as *mut u16;
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.write_volatile(value) }
+    }
+
+    fn write_pci_u32(
+        &self,
+        segment: u16,
+        bus: u8,
+        device: u8,
+        function: u8,
+        offset: u16,
+        value: u32,
+    ) {
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_mut_ptr() as *mut u32;
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.write_volatile(value) }
+    }
+}
+
+struct Acpi {
+    rsdp: AcpiTables<LocalAcpiHandler>,
+    aml_context: AmlContext,
+}
+
+/// irq model used in ACPI
+#[allow(dead_code)]
+enum X86IrqModel {
```

Comment:
The variant name causes `cargo clippy` warnings: 
```
warning: name `PIC` contains a capitalized acronym
   --> modules/axhal/src/platform/pc_x86/acpi.rs:206:5
    |
206 |     PIC,
    |     ^^^ help: consider making the acronym lowercase, except the initial letter (notice the capitalization): `Pic`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms
    = note: `#[warn(clippy::upper_case_acronyms)]` on by default
```

Reply:
OK. I changed the enum names to "UpperCamelCase".